### PR TITLE
Virtual network list page

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Add a page listing the virtual networks.

## GUI diff

Before:

localhost_8080_rhn_manager_systems_details_virtualization_guests_1000010000

After:

suma-virt-networks

- [X] **DONE**

## Documentation
- [Documentation PR](uyuni-project/uyuni-docs/pull/372)

- [x] **DONE**

## Test coverage
- Cucumber tests were added

- [X] **DONE**

## Links

Fixes: SUSE/spacewalk#11806
**Based  on PR #2352**

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/suse/manager/webui/controllers/virtualization/AbstractVirtualizationController.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+     * @return the server Id
+     *
+     * @throws NotFoundException if the server id has an invalid format or if the sid parameter can't be found
+     */
+    protected long getServerId(Request request) throws NotFoundException {
+        try {
+            return Long.parseLong(request.params("sid"));
+        }
+        catch (NumberFormatException e) {
+            throw new NotFoundException();
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply java/code/src/com/suse/manager/webui/controllers/virtualization/AbstractVirtualizationController.java]: done

**Comment by cbosdo** [comment]
got rid of the Salt pools-event related commits. Not depending on pending Salt PR anymore

**Comment by juliogonzalez** [comment]
Dear developer:

A branch for SUSE Manager 4.1 is now created on the internal repository, and all package SPEC files in the master branch of this repository are now updated to be '4.2.0' version.

All developers, regardless of being SUSE employee or not, need to keep the following things in mind:

1. Please rebase this PR against `master`branch and fix any conflicts.
2. If this PR includes schema upgrades, change the directory for the migration scripts to the **correct one**. At this precise moment it is `schema/spacewalk/upgrade/susemanager-schema-4.2.0-to-susemanager-schema-4.2.1` but it could be a different one if this PR is still open after Uyuni 2020.08 is released.

SUSE employees must also keep in mind:

1. If you merge the PR here, you most likely want to backport it to the new `Manager-4.1` branch.
2. You will need to **adapt** the PR when backporting for at least the changelogs and schema migrations (if there are on your PR).

**Comment by cbosdo** [comment]
Closing since the global PR has been merged

**Pull request closed without merging**